src/NativeScript/inlineFunctions.js

```diff
@@ -14,6 +14,103 @@ Object.assign(global, {
     },
+
+    // Decorators support    
+    __decorate(decorators, target, key, desc) {
+        var c = arguments.length;
+        var r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc;
+        var d;
+
+        if (typeof global.Reflect === "object" && typeof global.Reflect.decorate === "function") {
+            r = global.Reflect.decorate(decorators, target, key, desc);
+        }
+        else {
+            for (var i = decorators.length - 1; i >= 0; i--) {
+                if (d = decorators[i]) {
+                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
+                }
+            }
+        }
+        return c > 3 && r && Object.defineProperty(target, key, r), r;
+    },
+    __param(paramIndex, decorator) {
+        return function (target, key) {
+            decorator(target, key, paramIndex);
+        }
+    },  
+
+    // Decorators  
+    ObjC(name) {
+        var additionalParams = [];
+        var hasName = (name !== undefined && typeof name === "string");
+        var argsShift = hasName ? 1 : 0;
+        for (var i = argsShift; i < arguments.length; i++) {
+            additionalParams[i - argsShift] = arguments[i];
+        }
+
+        return function (target, propertyKey, descriptor) {
+            if (propertyKey === undefined) { // Class decorator
+                var implementation = {};
+                var baseTarget = Object.getPrototypeOf(target);
+                
+                for (var prop in target.prototype) {
+                    if (target.prototype.hasOwnProperty(prop)) {
+                        implementation[prop] = target.prototype[prop];
+                    }
+                }
+
+                var newName = hasName ? name : target.name;                
+                var extended = baseTarget.extend(implementation, {
+                    name: newName,
+                    protocols: additionalParams.length > 0 ? additionalParams : undefined,
+                    exposedMethods: target.ObjCExposedMethods
+                });
+                extended.name = newName;
+
+                // Add static members
+                for (var prop in target) {
+                    if (target.hasOwnProperty(prop) && prop !== "ObjCExposedMethods") {
+                        extended[prop] = target[prop];
+                    }
+                }
+
+                return extended;
+            }
+
+            // Method Decorator
+            if (!target.constructor.ObjCExposedMethods) {
+                target.constructor.ObjCExposedMethods = {};
+            }
+            if (!target.constructor.ObjCExposedMethods[propertyKey]) {
+                target.constructor.ObjCExposedMethods[propertyKey] = {};
+            }
+            target.constructor.ObjCExposedMethods[propertyKey].returns = additionalParams[0] || interop.types.void;
+
+            if (hasName && name !== propertyKey) {
```

Comment:
I don't think either point is valid and here is why:
1. The decorators provide TypeScript friendly API over existing functionality or provide shorthand for frequently used patterns. I needed the special thing and as a TypeScript developer I think it is convenient
2. As a TypeScript developer I will probably check the methods of the base class when I am picking the TypeScript friendly name, just like I would when I am adding any other method that may unintentionally override a base class method.